pyproject.toml

```diff
@@ -26,12 +27,19 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.8,<3.12"
+python = ">=3.8,<3.13"
 click = ">=7.2,<=9.0"
 click-default-group = "^1.2.2"
-numpy = "^1.19"
+numpy = [
+    { version = "^1.19,<1.25", python = "<3.9" },
+    { version = "^1.19", python = ">=3.9,<3.12" },
+    { version = "^1.26", python = ">=3.12" },
+]
 Pillow = ">=9.1,<10.0"
-scipy = "^1.7.3"
+scipy = [
+    { version = "^1.7.3,<1.11", python = "<3.9" },
+    { version = "^1.7.3", python = ">=3.9" }
+]
```

Comment:
Is there a reason why these constraints are specified with respect to the Python version?

Reply:
The may not be the best constraints, but this was needed to make Poetry happy with running `poetry.lock`.

I think this is needed because there is a version of both packages that both support Python 3.8 and 3.12.

Reply:
Hum interesting, I think we might want to have it working without this workaround, as this will affect users installing `manim` as a published package and are not dealing with Poetry at all (i.e. `poetry.lock` is not involved when installing from PyPI)

Reply:
This should not affect users, since:

- for NumPy, `v1.25` is the first version that does not support Python 3.8, and `v.1.26` is the earliest version supporting Python 3.18;
- for SciPy, the same logic applies.

Reply:
Made a typo in my previous comment:

> there is **no** version...

Reply:
I'll take a deeper look at this, feels weird having to do this as usually these constraints are solved by the build tool. Anyway, we will soon require `>=1.21` to use `numpy.typing`

Reply:
So I've checked out `main`, deleted the lock file, bumped `python` to `python = ">=3.8,<3.13"` (as you did), and ran `poetry lock` again without issues. Have you done something similar?

Reply:
Yes this is what I did (except deleting the lock file), but it didn't work...
Just ran the commands again, and it now works. I have updated the files accordingly.

Reply:
Nice!